src/build.rs

```diff
@@ -406,10 +407,33 @@ pub struct LibraryCApiConfig {
     pub version: Version,
     pub install_subdir: Option<String>,
     pub versioning: bool,
+    pub version_suffix_components: Option<u8>,
     pub import_library: bool,
     pub rustflags: Vec<String>,
 }
 
+impl LibraryCApiConfig {
+    pub fn sover(&self) -> anyhow::Result<String> {
+        let major = self.version.major;
+        let minor = self.version.minor;
+        let patch = self.version.patch;
+
+        let sover = match self.version_suffix_components {
+            None => match (major, minor, patch) {
+                (0, 0, patch) => format!("0.0.{patch}"),
+                (0, minor, _) => format!("0.{minor}"),
+                (major, _, _) => format!("{major}"),
+            },
+            Some(1) => format!("{major}"),
+            Some(2) => format!("{major}.{minor}"),
+            Some(3) => format!("{major}.{minor}.{patch}"),
+            Some(num) => anyhow::bail!("Unexpected number of suffix components: {num}"),
```

Comment:
If you want to go the extra mile, you should put the validation when parsing the value instead.

Reply:
Good idea, done!